apps/ui-patterns/src/app/UIPatternsComplexWizard/UIPatternsComplexWizard.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * UIPatternsComplexWizard.tsx
+ *
+ * Example of how to create a wizard that supports extra values passed in (possibly via hook)
+ * that are used in initial value generation and custom step validation. This also includes
+ * examples of how to use Field Arrays in Formik
+ */
+import React, { useCallback, useMemo } from 'react';
+import useWizardValues, { UIWizardProvider } from './useWizardValues';
+import { UIWizardStep, UIWizardSchema, UIWizardExtra } from '.';
+import { Wizard, WizardStep } from '@tacc/core-wrappers';
+import wizardSteps from './steps';
+
+export const UIPatternsComplexWizardRender: React.FC<{
+  wizardSteps: Array<UIWizardStep>;
+}> = ({ wizardSteps }) => {
+  const { values, add, extra } = useWizardValues();
+
+  // The submit button rendered by the Wizard should add values to the
+  // wizard value context
+  const formSubmit = useCallback(
+    (value: Partial<UIWizardSchema>) => {
+      console.log('Adding', value);
+      add(value);
+    },
+    [add]
+  );
+
+  // Map Array of UIWizardSteps into an array of @tacc/core-wrappers/WizardStep
+  const steps: Array<WizardStep<UIWizardSchema>> = useMemo(() => {
+    return wizardSteps.map((jobStep) => {
+      const { generateInitialValues, validateThunk, ...stepProps } = jobStep;
+      return {
+        // Call the step's custom initial value generator to pass in the extra hook values
+        initialValues: generateInitialValues({ values, extra }),
+        // generate a validation function from the UIWizardSteps's validateThunk, given the current extra hook values
+        validate: validateThunk ? validateThunk(extra) : undefined,
+        ...stepProps,
+      };
+    });
+  }, [wizardSteps, extra]);
+
+  return (
+    <Wizard steps={steps} memo={`${extra.memo}`} formSubmit={formSubmit} />
+  );
+};
+
+const UIPatternsComplexWizard: React.FC = () => {
+  const defaultValues: Partial<UIWizardSchema> = {
+    fieldArray: [{ name: 'item one', include: true }],
+    fieldArrayOfArrays: [
+      {
+        name: 'outer item',
+        fieldArray: [{ name: 'inner item one', include: true }],
+      },
+    ],
+  };
+  const extra: UIWizardExtra = {
+    extraOne: 'extra value one',
+    extraTwo: 'extra value two',
+    memo: 'wizard.memo',
+  };
+  return (
+    <UIWizardProvider value={{ defaultValues, extra }}>
+      <UIPatternsComplexWizardRender wizardSteps={wizardSteps} />
+    </UIWizardProvider>
+  );
+};
+
+export default UIPatternsComplexWizard;
```

Comment:
**Complex Wizard**
If I unfurl "more values inside", then the button disappears, but still works.

https://user-images.githubusercontent.com/62723358/175393964-450fc79d-5aa6-4e60-9b49-d77453def5cf.mov

Reply:
Fixed permanently, actually. Thanks. I remember this class name now.

The `collapse` was old name that I did not support when I brought Cortal icons to core-styles. The `contract` is the new name in altest cortal icons. _I do retain legacy support for it in Portal, but should drop it and update usage._